### PR TITLE
[core] Filesystem catalog supports setting case-sensitive option.

### PR DESCRIPTION
**Pull request by zhuangchong**
### Purpose

 Filesystem catalog supports setting case-sensitive option.

### CODE REVIEW

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/kafka/KafkaSyncDatabaseAction.java [hunk 1]
```diff
@@ -183,21 +180,9 @@ public void build() throws Exception {
...
-                tablePrefix.equals(tablePrefix.toLowerCase()),
-                String.format(
-                        "Table prefix [%s] cannot contain upper case in case-insensitive catalog.",
-                        tablePrefix));
-        checkArgument(
-                tableSuffix.equals(tableSuffix.toLowerCase()),
-                String.format(
-                        "Table suffix [%s] cannot contain upper case in case-insensitive catalog.",
-                        tableSuffix));
+        catalog.validateCaseInsensitive("Database", database);
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/kafka/KafkaSyncDatabaseAction.java:183]: This should be just a util method... Not a public method.

**Comment by JingsongLi** [comment]
Thanks @zhuangchong , left comment about public method in `Catalog`.

**Comment by zhuangchong** [comment]
> Thanks @zhuangchong , left comment about public method in `Catalog`.

Thanks @JingsongLi ,  done.

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**